### PR TITLE
Update segmentation documentation

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary
1. Update docs
2. Provide new benchmark results
3. Minor fixes for predict mode

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by kprokofi** [comment]
waiting benchmark results...

File: docs/source/guide/explanation/algorithms/segmentation/semantic_segmentation.rst [hunk 1]
```diff
@@ -55,31 +48,32 @@ Models
...
-
-Use `SegNext <https://arxiv.org/abs/2209.08575>`_ model which can achieve superior perfomance while preserving fast inference and fast training.
-
-In the table below the `Dice score <https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient>`_ on some academic datasets using our :ref:`supervised pipeline <semantic_segmentation_supervised_pipeline>` is presented. We use 512x512 image crop resolution, for other hyperparameters, please, refer to the related recipe. We trained each model with single Nvidia GeForce RTX3090.
++--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------+-----------------+-----------------+-----------------+
+| Recipe Path                                                                                                                                                                          | Complexity (GFLOPs) | Model size (M)  | FPS (GPU)       | iter time (sec) |
```
Comment (sovrasov) [inline docs/source/guide/explanation/algorithms/segmentation/semantic_segmentation.rst:52]: How is GPU FPS measured?

Comment (kprokofi) [reply docs/source/guide/explanation/algorithms/segmentation/semantic_segmentation.rst:52]: image

Comment (sovrasov) [reply docs/source/guide/explanation/algorithms/segmentation/semantic_segmentation.rst:52]: Is `self.model = torch.compile(self.model)` applied at some point? I see it in the `OTXModel.setup()` hook, but not sure, if it's called in that context

**Comment by kprokofi** [comment]
image

Comparison between new OTX2.0 (decoupled) and OTX1.5 (with masks annotation (without to/from polygons transformation) for fair comparison)

File: docs/source/guide/explanation/algorithms/segmentation/semantic_segmentation.rst (hunk 1 above)
Comment (kprokofi) [reply docs/source/guide/explanation/algorithms/segmentation/semantic_segmentation.rst:52]: I used this snippet of code inside Engine just before training (trainer.fit()). So, if we use torch.compile, it should be applied

Comment (eunwoosh) [reply docs/source/guide/explanation/algorithms/segmentation/semantic_segmentation.rst:52]: As Vlad said, `torch.compile` exists in `OTXModel.setup()` but it's executed only if `self.torch_compile` is `True`. It seems that default value of it is set as `False` and any default recipe doesn't change that value, so `torch.compile` isn't executed as default.

**Review by jaegukhyun — APPROVED** [review]

**Review by sovrasov — APPROVED** [review]

**Pull request merged**